cogs/archive.py

```diff
@@ -34,34 +33,66 @@ async def autocomplete_get_categories(ctx: TeXBotAutocompleteContext) -> Set[dis
 
-        The list of available selectable categories is unique to each member and is used in
-        any of the "archive" slash-command options that have a category input-type.
+        The list of categories only includes those that do not contain the word "archive".
         """
-        if not ctx.interaction.user:
+        try:
+            main_guild: discord.Guild = ctx.bot.main_guild
+        except BaseDoesNotExistError:
             return set()
 
-        try:
-            if not await ctx.bot.check_user_has_committee_role(ctx.interaction.user):
-                return set()
+        return {
+            discord.OptionChoice(name=category.name, value=str(category.id))
+            for category
+            in main_guild.categories
+            if "archive" not in category.name.lower()
+        }
+
+    @staticmethod
+    async def autocomplete_get_archival_categories(ctx: TeXBotAutocompleteContext) -> Set[discord.OptionChoice] | Set[str]:  # noqa: E501
+        """
+        Autocomplete callable that generates the set of categories to hold archived channels.
 
+        The list of categories only includes those that contain the word "archive".
+        These are the categories that channels are to be placed into for archiving.
+        It is assumed that the categories have the correct permission configuration.
+        """
+        try:
             main_guild: discord.Guild = ctx.bot.main_guild
-            interaction_user: discord.Member = await ctx.bot.get_main_guild_member(
-                ctx.interaction.user,
-            )
-        except (BaseDoesNotExistError, DiscordMemberNotInMainGuildError):
+        except BaseDoesNotExistError:
             return set()
 
         return {
             discord.OptionChoice(name=category.name, value=str(category.id))
             for category
             in main_guild.categories
-            if category.permissions_for(interaction_user).is_superset(
-                discord.Permissions(send_messages=True, view_channel=True),
+            if "archive" in category.name.lower()
+        }
+
+    @staticmethod
+    async def autocomplete_get_channels(ctx: TeXBotAutocompleteContext) -> Set[discord.OptionChoice] | Set[str]:  # noqa: E501
+        """
+        Autocpomplete callable that generates the set of channels that the user can archive.
+
+        The list of channels will include all types of channels except categories,
+        that have not been archived.
+        """
+        try:
+            main_guild: discord.Guild = ctx.bot.main_guild
+        except BaseDoesNotExistError:
+            return set()
+
+        return {
+            discord.OptionChoice(name=channel.name, value=str(channel.id))
+            for channel
+            in main_guild.channels
+            if (
+                not isinstance(channel, discord.CategoryChannel) and
+                (channel.category and "archive" not in channel.category.name.lower())
             )
         }
```

Comment:
This circumvents permissions as it allows non-committee users to view channels they would not normally be able to see